integration-tests/modules/__tests__/product/store/index.spec.ts

```diff
@@ -0,0 +1,332 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
```

Comment:
@riqwan I know you wrote a lot of tests here, but does it make sense to actually make the `integration-tests/api` tests work instead? If you feel like it will take a lot of time I'm fine with doing it later on, but we really need to try and clean up the tests

Reply:
2 points:

- I'd like these tests to be run on the CI, which it won't with the api tests.
- I don't get why we do that. Wouldn't it just be better to move over the test cases to modules and make it work for v2? We're not really writing more tests for v1, so we wouldn't be missing out much. We do the same everywhere else - v2 routes, v2 dashboard, why not the same here?

Reply:
@riqwan The idea is that, unless we intended a breaking change, the test scenarios in `api` are still valid and should work regardless if it is v1 or v2. Eg. for the product endpoints, we have very few breaking changes, and that gives us some confidence that we are compliant with v1, except for the intentional breaking changes.

But if you feel like that is not a valid point, then we should discuss and settle on a single approach, as currently we have a split between the two